code/modules/admin/verbs/listahelps.dm

```diff
@@ -0,0 +1,318 @@
+			usr << "	<b>Message:</b> [T.msg]"
+			usr << "	<b>Handling Admin:</b> [T.admin]"
+			usr << "	<b>Replied To:</b> [T.active]/<b><a href='?src=\ref[T];view_logs=\ref[T]'>(LOGS)</a></b>"
+			usr << "	<b>Resolved:</b> [T.resolved]"
+
+	if(upass == 0)
+		usr << "	None"
+
+/client/proc/createticket(var/player, var/message, var/uckey, var/mob)
+	var/datum/adminticket/A = new()
+	A.user = player
+	A.msg = message
+	A.uckey = uckey
+	A.permckey = uckey
+	A.permuser = A.user
+	admintickets += A
+	A.logs += "<b>ADMINHELP:</b> [A.permckey]([A.permuser]): [A.msg]"
+	A.mob = mob
+
+	var/index = 0
+	for(var/datum/adminticket/T in admintickets)
+		index++
+		T.ID = index
+		T.uID = "[T.permckey][T.ID]"
+
+/client/verb/resolveticketself()
+	set category = "Admin"
+	set name = "Resolve My Adminhelp"
+	set desc = "Resolve my own adminhelp"
+
+	var/pass = 0
+	var/datum/adminticket/ticket
+
+	for(var/datum/adminticket/T in admintickets)
+		if(T.permckey == src.ckey && T.resolved != "Yes")
+			T.resolved = "Yes"
+			ticket = T
+			pass = 1
+
+	switch(pass)
+		if(1)
+			src << "<b>You have resolved your current adminhelp.</b>"
+			message_admins("[src] has resolved his adminhelp (#[ticket.ID])")
+		if(0)
+			src << "<b>Error, you do not have any active adminhelps.</b>"
+
+/client/proc/resolvehandlingahelp()
+	set category = "Admin Help"
+	set name = "Resolve Handling Ahelp"
+	set desc = "Resolve my own adminhelp"
+
+	if(!check_rights(R_BAN))
+		src << "<font color='red'>Error: Only administrators may use this command.</font>"
+		return
+
+	var/count = 0
+
+	var/datum/adminticket/ticket
+
+	for(var/datum/adminticket/T in admintickets)
+		if(T.admin == src.ckey && T.resolved != "Yes")
+			count++
+			ticket = T
+
+	if(count >= 1)
+		usr << "<b>Adminhelp #[ticket.ID]([ticket.uID]) resolved.</b>"
+		message_admins("Adminhelp ID: #[ticket.ID]([ticket.uID]) was resolved by [usr.ckey]")
+		ticket.user << "<b>Your adminhelp (#[ticket.ID]) has been resolved by [usr.ckey]<b>"
+		ticket.user << 'sound/machines/twobeep.ogg'
+		ticket.resolved = "Yes"
+
+	if(count < 1)
+		usr << "<b>You are not currently handling any adminhelps!</b>"
+
+/datum/adminticket/proc/viewlogs(var/NuID, mob/user)
+	var/dat = "<h3>View Logs for ahelp [NuID]</h3>"
+	var/datum/adminticket/ticket
+
+	var/pass = 0
+
+	for(var/datum/adminticket/T in admintickets)
+		if(NuID == T.uID)
+			ticket = T
+			pass = 1
+
+	if(pass == 0)
+		src << "Error, log system not found for [NuID]... "
+		return
+
+	dat += "<table>"
+	for(var/text in ticket.logs)
+		dat += "<tr><td>[text]</td></tr>"
+	dat += "</table>"
+
+	var/datum/browser/popup = new(user, "ahelp logs", ticket.permuser, 500, 500)
+	popup.set_content(dat)
+	popup.open()
+
+/datum/adminticket/Topic(href, href_list)
+	if(href_list["view_logs"])
+		var/datum/adminticket/T = locate(href_list["view_logs"])
+		viewlogs(T.uID, usr)
+	if(href_list["resolve"])
+		var/datum/adminticket/T = locate(href_list["resolve"])
+		if(T.resolved == "Yes")
```

Comment:
instead of this you could use the ternary operator like message_admins("Adminhelp ID: #[T.ID]([T.uID]) was [T.resolved == "Yes" ? "unresolved" : "resolved"] by [usr.ckey]") etc